scripts/helper/create-topics.sh

```diff
@@ -1,47 +1,49 @@
 #!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
 
-# Create Kafka topic users, using appSA principal
-export KAFKA_LOG4J_OPTS="-Dlog4j.rootLogger=DEBUG,stdout -Dlog4j.logger.kafka=DEBUG,stdout" && kafka-topics \
-   --bootstrap-server kafka1:11091 \
-   --command-config /etc/kafka/secrets/appSA.config \
-   --topic users \
-   --create \
-   --replication-factor 2 \
-   --partitions 2 \
-   --config confluent.value.schema.validation=true
-
-# Create Kafka topics with prefix wikipedia, using connectorSA principal
-export KAFKA_LOG4J_OPTS="-Dlog4j.rootLogger=DEBUG,stdout -Dlog4j.logger.kafka=DEBUG,stdout" && kafka-topics \
-   --bootstrap-server kafka1:11091 \
-   --command-config /etc/kafka/secrets/connectorSA_without_interceptors_ssl.config \
-   --topic wikipedia.parsed \
-   --create \
-   --replication-factor 2 \
-   --partitions 2 \
-   --config confluent.value.schema.validation=true
-export KAFKA_LOG4J_OPTS="-Dlog4j.rootLogger=DEBUG,stdout -Dlog4j.logger.kafka=DEBUG,stdout" && kafka-topics \
-   --bootstrap-server kafka1:11091 \
-   --command-config /etc/kafka/secrets/connectorSA_without_interceptors_ssl.config \
-   --topic wikipedia.parsed.count-by-domain \
-   --create \
-   --replication-factor 2 \
-   --partitions 2
-export KAFKA_LOG4J_OPTS="-Dlog4j.rootLogger=DEBUG,stdout -Dlog4j.logger.kafka=DEBUG,stdout" && kafka-topics \
-   --bootstrap-server kafka1:11091 \
-   --command-config /etc/kafka/secrets/connectorSA_without_interceptors_ssl.config \
-   --topic wikipedia.failed \
-   --create \
-   --replication-factor 2 \
-   --partitions 2
-
-# Create Kafka topics with prefix WIKIPEDIA or EN_WIKIPEDIA, using ksqlDBUser principal
-for t in WIKIPEDIABOT WIKIPEDIANOBOT EN_WIKIPEDIA_GT_1 EN_WIKIPEDIA_GT_1_COUNTS
-do
-  export KAFKA_LOG4J_OPTS="-Dlog4j.rootLogger=DEBUG,stdout -Dlog4j.logger.kafka=DEBUG,stdout" && kafka-topics \
-     --bootstrap-server kafka1:11091 \
-     --command-config /etc/kafka/secrets/ksqlDBUser_without_interceptors_ssl.config \
-     --topic "$t" \
-     --create \
-     --replication-factor 2 \
-     --partitions 2
-done
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+KAFKA_CLUSTER_ID=$(curl -s --insecure --location --request GET 'https://localhost:8091/kafka/v3/clusters' --header 'Authorization: Basic c3VwZXJVc2VyOnN1cGVyVXNlcg==' | jq -r '.data[0].cluster_id')
+
+echo "cluster_id: ${KAFKA_CLUSTER_ID}"
+
+curl -s --insecure --location --request POST "https://localhost:8091/kafka/v3/clusters/${KAFKA_CLUSTER_ID}/topics" \
+--header 'Content-Type: application/json' \
+--header 'Authorization: Basic c3VwZXJVc2VyOnN1cGVyVXNlcg==' \
+--data-binary @<(jq -n --arg topic_name users --arg confluent_value_schema_validation "true" -f ${DIR}/topic.jq) | jq
+
+curl -s --insecure --location --request POST "https://localhost:8091/kafka/v3/clusters/${KAFKA_CLUSTER_ID}/topics" \
```

Comment:
For comparison (and really promoting if these examples can be or should be consistent), the cp-demo docs shows an example as follows:

```
      docker-compose exec restproxy curl -X POST \
         -H "Content-Type: application/json" \
         -H "accept: application/json" \
         -d "{\"topic_name\":\"dev_users\",\"partitions_count\":64,\"replication_factor\":2,\"configs\":[{\"name\":\"cleanup.policy\",\"value\":\"compact\"},{\"name\":\"compression.type\",\"value\":\"gzip\"}]}" \
         --cert /etc/kafka/secrets/mds.certificate.pem \
         --key /etc/kafka/secrets/mds.key \
         --tlsv1.2 \
         --cacert /etc/kafka/secrets/snakeoil-ca-1.crt \
         -u appSA:appSA \
         "https://kafka1:8091/kafka/v3/clusters/${KAFKA_CLUSTER_ID}/topics" | jq
```

There are differences between what the cp-demo docs show and what is written here.  Can we think about if it makes sense to synchronize them (e.g. the user/auth, `--insecure` versus not`, passing in the certs...etc)

Reply:
Script code is now mostly consistent with this, remaining differences:
- `-d` data, ours is dynamically generated for each call from a `jq` template
- We aren't sending a `key` ... we rely on Basic auth ... above sets both `key` and `user`, should probably pick one.
- Example sets `--tls1.2` forcing a less-secure-than-available protocol (1.3 is best, available) - doc-bug?

Reply:
@javabrett 

> We aren't sending a key ... we rely on Basic auth ... above sets both key and user, should probably pick one.

When removing `key`, the command errors out with: `parse error: Invalid numeric literal at line 1, column 5`.  Does this happen in your env and do you know why?

> Example sets --tls1.2 forcing a less-secure-than-available protocol (1.3 is best, available) - doc-bug?
 
IIRC `v1.2` was specified NOT instead of `v1.3` but instead of `v1.1`.  Is your recommendation to remove the version completely from the command examples in the docs, or to explicitly set `v1.3`?

Reply:
> Does this happen in your env and do you know why?

IIUC this is because user `appSA` has only been granted `ResourceOwner` on some topics, so might not have grant to create a topic?  I might be misreading the grants.

Changing to `-u superUser:superUser` with no keys or certs works with Basic auth.  Maybe it's more correct to use the certs anyway?

Reply:
> IIRC v1.2 was specified NOT instead of v1.3 but instead of v1.1. Is your recommendation to remove the version completely from the command examples in the docs, or to explicitly set v1.3?

@ybyzek you are correct here, my error ... `--tlsv1.2` sets a minimum, so TLS v1.3 is typically being negotiated anyway, preventing TLS 1.1 or worse only.

I'll update the PR to include the same guide-rails.

Reply:
Added `--tlsv1.2` to `curl` for create topic.